dronegen/container_image_products.go

```diff
@@ -76,6 +130,141 @@ func NewTeleportOperatorProduct(cloneDirectory string) *Product {
 	}
 }
 
+func getTeleportSetupSteps(productName, workingPath, downloadURL string) ([]step, string) {
+	downloadDockerfileStep, dockerfilePath := downloadTeleportDockerfileStep(productName, workingPath, downloadURL)
```

Comment:
If we are checking out the source do we need to download the dockerfile?

Reply:
Yes and no. When the pipeline runs in the context of a branch that is _not_ the one it's building then it needs to download the version-specific dockerfile. For example, when triggered by a cron job the pipeline runs in the context of the master branch, but the cron jobs update images built from other specific branches (`branch/v11`, `branch/v10`, etc.).

I can make this step only run when strictly needed however IMO it's a premature optimization that would add complexity and not gain us much. Do you think I should implement that change?